File: Python/7py 2/greet_developers.py
"""
|--------------------------------------------------------------------------
| Coding Meetup #2 - Higher-Order Functions Series - Greet developers
|--------------------------------------------------------------------------
|
"""

def greet_developers(lst):
   for i in range(len(lst)):
      lst[i]['greeting'] = f"Hi {lst[i]['firstName']}, what do you like the most about {lst[i]['language']}?"
   return lst


def greet_developers(lst): 
   return [ {**dev, 'greeting': f"Hi {dev['firstName']}, what do you like the most about {dev['language']}?"} for dev in lst ]


def greet_developers(lst):
   for i in lst:
      i.update({"greeting": "Hi {firstName}, what do you like the most about {language}?".format(**i)})
   return lst


def greet_developers(lst): 
   return [dict(n,**{'greeting':f"Hi {n['firstName']}, what do you like the most about {n['language']}?"}) for n in lst]


list1 = [
   { 
      'firstName': 'Sofia',
      'lastName': 'I.',
      'country': 'Argentina',
      'continent': 'Americas',
      'age': 35, 
      'language': 'Java' 
   },
   { 
      'firstName': 'Lukas',
      'lastName': 'X.',
      'country': 'Croatia',
      'continent': 'Europe',
      'age': 35, 
      'language': 'Python' 
   },
   { 
      'firstName': 'Madison',
      'lastName': 'U.',
      'country': 'United States',
      'continent': 'Americas',
      'age': 32, 
      'language': 'Ruby' 
   } 
]
print(greet_developers(list1))

File: PHP/7php 1/greet_developers.php
<?php
/*
|--------------------------------------------------------------------------
| Coding Meetup #2 - Higher-Order Functions Series - Greet developers
|--------------------------------------------------------------------------
|
*/

function greet_developers($a) {
   for ($i=0; $i < count($a); $i++) { 
      $a[$i]['greeting'] = "Hi {$a[$i]['first_name']}, what do you like the most about {$a[$i]['language']}?";
   }
   return $a;
}


function greet_developers($a) {
   foreach($a as $key => $val){
      $a[$key]['greeting'] = 'Hi '.$val['first_name'].', what do you like the most about '.$val['language'].'?';
   }
   return $a;
}


function greet_developers($a) {
   return array_map(function($v) {
      $v['greeting'] = "Hi {$v['first_name']}, what do you like the most about {$v['language']}?";
      return $v;
   }, $a);
}


function greet_developers($a) {
   foreach ($a as &$p) {
      $p["greeting"] = "Hi " . $p["first_name"] . ", what do you like the most about " . $p["language"] . "?";
   }
   return $a;
}


$list1 = [
   [
      "first_name" => "Sofia",
      "last_name" => "I.",
      "country" => "Argentina",
      "continent" => "Americas",
      "age" => 35,
      "language" => "Java"
   ],
   [
      "first_name" => "Lukas",
      "last_name" => "X.",
      "country" => "Croatia",
      "continent" => "Europe",
      "age" => 35,
      "language" => "Python"
   ],
   [
      "first_name" => "Madison",
      "last_name" => "U.",
      "country" => "United States",
      "continent" => "Americas",
      "age" => 32,
      "language" => "Ruby"
   ]
];
print_r(greet_developers($list1));